packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -414,7 +421,23 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
     private process(message: ISequencedDocumentMessage, local: boolean, localOpMetadata: unknown) {
         this.verifyNotClosed(); // This will result in container closure.
         this.emit("pre-op", message, local, this);
-        this.processCore(message, local, localOpMetadata);
+
+        // Do not measure for local ops, because callbacks are not invoked for them in this codepath
+        // and including them in the measurements would skew the numbers.
+        if (local) {
+            this.processCore(message, local, localOpMetadata);
+        } else {
+            ExecuteWithAggregatedTelemetry(
```

Comment:
> We already have that combined time (not per DDS, but in general).

I imagine you're referring to the `GetDeltas_OpProcessing` data? If so, @anthony-murphy raised the concern that it has some caveats because in a scenario where a datastore/dds is not loaded yet, the duration reported in those events would just be that of putting the op in a local queue for later processing when the datastore/dds is loaded, not really dds-processing nor dds-callbacks. Tracking time spent in `processCore` at least would guarantee that the op is actually being processed, and we could get an estimate of app-incurred-duration VS fluid-incurred-duration (at the DDS level) by subtracting the two things for which I added tracking (`processCore`, and the `emit` calls).

Tony and I also discussed making these measurements granular at the DDS level (maybe even to the specific event raised by the DDS) and the concern was that since users can implement their own dds/events and give them their own names, we needed to be careful on what we put in the telemetry event. Neither of those two things would be safe per-se; if we want data with that granularity, maybe we hash the dds type or the event name, so we can at least look at the slices in Kusto? We would know the hashes of the framework dds so we could do some cross-referencing there.

Reply:
@markfields what is the current right way to tag data, like the dds type here?

Reply:
if we add dds ype, i'd put it on the base logger, rather than the event, as if we have it, we should just have it for all dds logs

Reply:
If these values come from the DDS implementation, then the default is to use `CodeArtifact`.

To be transparent, this is a catch-all bucket that is basically ignored today (our Loop partners log the tagged values plainly) based on the premise that 1P hosts don't actually collect data involving 3P DDSes or DataStores. But someday this could change so we wanted to have a concept of it in our code, even if it's pretty rough. cc @jvoels -- I saw you mention this exact scenario (and what a challenge it is from telemetry POV...), please keep in mind any requirements you'd have on FF logging tags to be able to disect these values when scenarios like this materialize.